src/clib/lib/res_util/subst_list.cpp

```diff
@@ -537,10 +537,10 @@ static char *trim_string(char *str) {
     return str;
 }
 
-void subst_list_add_from_string(subst_list_type *subst_list,
-                                const char *arg_string_orig) {
+const char *subst_list_add_from_string(subst_list_type *subst_list,
+                                       const char *arg_string_orig) {
     if (!arg_string_orig)
-        return;
+        return "No argument string provided";
```

Comment:
I am bit worried (perhaps unnecessarily) about the lifetime of these string as they are passed to the python runtime via cwrap. Are they appropriately copied and guaranteed to not be freed before that time?

Reply:
I believe @valentin-krasontovitsch investigated using pybind and raising an exception in cpp and catching it in python. This is likely a better alternative.

Reply:
ya i can totally help with that pattern if you should choose that way

Reply:
I concur. We can discuss this briefly at standup. Thanks for the input.